docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-22T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000,
+  "reason": "The type 'typeA' is not configured."
+}
+```
+
+### Successful flow
+
+The status transition: `init` -> `executing` -> `uploading` -> `successful`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
+  alt if necessary
+    Mapper->>Cloud: Status: uploading
+  end
+  Mapper->>Cloud: Upload log file [HTTP]
+  Cloud-->> Mapper: Return the http endpoint of the uploaded file [HTTP]
+  alt if necessary
+    Mapper->>Cloud: Status: successful
+    Mapper->>Cloud: Cloud-specific something if any
+  end
+```
+
+### Failed flows
+
+a. Failure on the plugin side
+
+The status transition: `init` -> `executing` -> `failed`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log (FAILED!)
+  Plugin->>Mapper: Status: failed with reason
+  alt if necessary
+    Mapper->>Cloud: Status: failed with reason
+  end
+```
+
+b. Failure on uploading log (mapper side)
+
+The status transition: `init` -> `executing` -> `uploading` -> `failed`
+
+```mermaid
+sequenceDiagram
+  participant Cloud
+  participant Mapper
+  participant Plugin
+
+  Cloud->>Mapper: Cloud-specific Log file upload request (Status: init)
+  Mapper->>Plugin: Cloud-agnostic Log file upload request (Status: init)
+  Plugin->>Mapper: Status: executing
+  alt if necessary
+    Mapper->>Cloud: Status: executing
+  end
+  Plugin->>Plugin: Extract log
+  Plugin->>Mapper: File transfer [HTTP]
+  Plugin->>Mapper: Status: uploading
+  alt if necessary
+    Mapper->>Cloud: Status: uploading
+  end
+  Mapper->>Cloud: Upload log file [HTTP] (FAILED!)
+  Cloud-->> Mapper: Return the http endpoint of the uploaded file [HTTP] (FAILED CODE!)
+  alt if necessary
+    Mapper->>Cloud: Cloud-specific Status failed
+  end
+```
+
+## Usage
+
+TBD
+
+## Logging
+
+The `tedge-log-plugin` reports progress and errors to the OS journal which can be retrieved using `journalctl`.
+
+
+## Not clear / undecided
+
+* Renaming the toml file from `c8y-log-plugin.toml` to `tedge-log-plugin.toml`.
+* Supported operation. Create automatically `c8y_LogfileRequest` as it is now? Or ask user to create it?
```

Comment:
I suspect this will be influence by the new proposal to support a declaration of thin-edge agnostic command support (e.g. by a component publishing a retained mqtt message to `te/<indentifier>/cmd/<cmd_type>`). The cloud mappers could register to each of these command meta topics to see which command type are supported on the agnostic level, and decide if it knows how to map them to the cloud specific name, e.g. `log_upload` --> `c8y_LogFileRequest`...so I'm not sure what value having the physical file has, though we'll need to figure out how to handle it in the transitional phase whilst we're still finalizing the command meta registration details.

Reply:
It should be the mapper that creates this file for its own tracking when it receives the supported log types message from the plugin.

Reply:
I would even consider to more use such files.

Indeed, the `log_upload` capability will be published as a retained message on `te/device/main///cmd/log_upload`
for the main device and on `te/device/child-xyz///cmd/log_upload` for a child device. A mapper has to subscribe to all these retained messages and to translate them appropriately for the cloud.

So a `c8y_LogfileRequest` file can be used for caching purpose but not to detect that the capability is available or not.